AgentInterfaces/AssetDetail.cs

```diff
@@ -22,5 +22,12 @@ public class AssetDetail
         /// For developer testing, specify a file path accessible locally.
         /// </summary>
         public string LocalFilePath { get; set; }
+
+        public AssetDetail ToRedacted()
+        {
+            AssetDetail clone = (AssetDetail)this.MemberwiseClone();
```

Comment:
how can we verify that we don't forget to redact a property we add later? any chance we can have a test to verify that?

Reply:
I don't think adding a test for this would be possible--there's no way to automate knowing what fields need to be redacted. Not even an automated credential scanning library could do that in a unit test, because the library wouldn't flag the new property it unless we filled it with an actual credential